shell/platform/windows/flutter_windows_view.cc

```diff
@@ -80,19 +90,22 @@ uint32_t FlutterWindowsView::GetFrameBufferId(size_t width, size_t height) {
 void FlutterWindowsView::OnWindowSizeChanged(size_t width, size_t height) {
   // Called on the platform thread.
   std::unique_lock<std::mutex> lock(resize_mutex_);
+  if (!IsValidResize(resize_target_width_, resize_target_height_, width,
+                     height)) {
+    return;
+  }
+
   resize_status_ = ResizeState::kResizeStarted;
   resize_target_width_ = width;
   resize_target_height_ = height;
   SendWindowMetrics(width, height, binding_handler_->GetDpiScale());
 
-  if (width > 0 && height > 0) {
```

Comment:
Hm, I assumed from the bug description that we were blocking on an update that never came, but it looks like that already shouldn't have been the case. What exactly is causing the hang then?

Reply:
The hang is caused because we don't ask for an FBO when the onscreen surface that is held transitively by the rasterizer is the same size as the requested size.

The resizing logic here assumes that every request to `OnWindowSizeChanged` will result in a new FBO being requested which is violated when we minimize-restore as we don't destroy the on-screen surface when that happens.

I will follow your recommendation and link to https://github.com/flutter/engine/pull/23990#discussion_r565514414 with a pointer that we likely want to destroy the on-screen surface when the application is minimized which would resolve this. For now, I will special case this in the resizing logic.

Reply:
Oh, we're already keeping the old surface around? I thought it was controlled directly by the size we got from the OS.

If we have a surface at the old size still, then I think I'm fine with your approach here, as long as you comment as to why.